app/src/main/kotlin/edu/stanford/bdh/engagehf/messages/HealthSummaryService.kt

```diff
@@ -0,0 +1,60 @@
+package edu.stanford.bdh.engagehf.messages
+
+import android.content.Context
+import android.content.Intent
+import android.os.Environment
+import androidx.core.content.FileProvider
+import dagger.hilt.android.qualifiers.ApplicationContext
+import edu.stanford.spezi.core.logging.speziLogger
+import edu.stanford.spezi.core.utils.MessageNotifier
+import java.io.File
+import java.io.FileOutputStream
+import javax.inject.Inject
+
+class HealthSummaryService @Inject constructor(
+    private val healthSummaryRepository: HealthSummaryRepository,
+    private val messageNotifier: MessageNotifier,
+    @ApplicationContext private val context: Context,
+) {
+    private val logger by speziLogger()
+
+    companion object {
+        const val FILE_NAME = "health_summary.pdf"
+        const val MIME_TYPE_PDF = "application/pdf"
+    }
+
+    suspend fun generateHealthSummaryPdf() {
+        val pdfByteArrayResult =
+            healthSummaryRepository.findHealthSummaryByUserId()
+        pdfByteArrayResult.onSuccess {
+            val savePdfToFile = savePdfToFile(it)
+            val pdfUri = FileProvider.getUriForFile(
+                context,
+                "${context.packageName}.provider",
+                savePdfToFile
+            )
+            val intent = Intent(Intent.ACTION_VIEW)
+            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            intent.setDataAndType(pdfUri, MIME_TYPE_PDF)
+            context.startActivity(intent)
+        }.onFailure {
+            messageNotifier.notify("Failed to generate Health Summary")
+        }
+    }
+
+    private fun savePdfToFile(pdfBytes: ByteArray): File {
+        logger.i { "PDF size: ${pdfBytes.size}" }
+        val storageDir =
+            Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)
+        if (!storageDir.exists()) {
+            storageDir.mkdirs()
+        }
+        val pdfFile = File(storageDir, FILE_NAME)
+        FileOutputStream(pdfFile).use { fos ->
+            fos.write(pdfBytes)
```

Comment:
the current situation is that the file always has the same name and is therefore continuously overwritten. is there a way how we can recognise whether our local file is still up to date, @pauljohanneskraft @PSchmiedmayer ? i would almost prefer it if we integrate a mechanism as you suggested @eldcn, which makes it possible not to have to load it constantly because the file has 8-9 mb so we can save a little traffic